### PR TITLE
Greyscale sombrero and shamebrero

### PR DESCRIPTION
**Pull request by ninjanomnom**
image

Mostly I did this in the process of making a step by step guide for GAGS [on hackmd](https://hackmd.io/@tgstation/GAGS-Walkthrough). No player facing changes really, though an admin could more easily change the colors now if they wish.

### CODE REVIEW

File: code/datums/greyscale/json_configs/sombrero_base.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+            "type": "icon_state",
+            "icon_state": "cloth",
+            "blend_mode": "overlay",
+            "color_ids": [ 1 ]
+        },
+        {
+            "type": "icon_state",
+            "icon_state": "ornamentation",
+            "blend_mode": "overlay",
+			"color_ids": [ 2 ]
```
Comment (celotajstg) [inline code/datums/greyscale/json_configs/sombrero_base.json:13]: indentation

File: code/datums/greyscale/json_configs/sombrero_lefthand.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+			"color_ids": [ 1, 2 ]
+		}
+	],
+    "shamebrero": [
+        {
+			"type": "reference",
+			"reference_type": "/datum/greyscale_config/sombrero/base_lefthand",
+			"blend_mode": "overlay",
+			"color_ids": [ 1, 2 ]
+		},
```
Comment (celotajstg) [inline code/datums/greyscale/json_configs/sombrero_lefthand.json:16]: same here and in some other places, would be good to re-check

File: code/modules/clothing/head/misc.dm [hunk 3]
```diff
@@ -288,20 +288,29 @@
 
 	dog_fashion = /datum/dog_fashion/head/sombrero
 
+	greyscale_config = /datum/greyscale_config/sombrero
+
+/obj/item/clothing/head/sombrero/Initialize()
+	. = ..()
+	if(greyscale_colors)
```
Comment (celotajstg) [inline code/modules/clothing/head/misc.dm:295]: no need for this, see: https://github.com/tgstation/tgstation/pull/58808/files#diff-c8ab5fbc20de60e202b839834b039649cbb69a1c4b99b27a5e467f3889442ccdR38

tl;dr - just assign greyscale_config_worn, greyscale_config_inhand_left and greyscale_config_inhand_right variables.

**Review by celotajstg — CHANGES_REQUESTED** [review]

**Review by Twaticus — APPROVED** [review]
